src/librustc_mir/build/matches/mod.rs

```diff
@@ -541,7 +538,7 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
             PatternKind::Deref { ref subpattern } => {
                 self.visit_bindings(subpattern, pattern_user_ty.deref(), f);
             }
-            PatternKind::AscribeUserType { ref subpattern, ref user_ty, user_ty_span } => {
+            PatternKind::AscribeUserType { ref subpattern, ref user_ty, user_ty_span, .. } => {
```

Comment:
I would rather change `..` to `variance: _` and add a comment below:

---

Note that the variance doesn't apply here, as we are tracking the effect of `user_ty` on any bindings contained with `subpattern`.

Reply:
Changed.